storage-access.bs

```diff
@@ -162,7 +164,10 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>ha
 
     ISSUE: "same authority" here is a placeholder for a future concept that allows user agents to perform [=same site=] checks while adhering to additional security aspects such as the presence of a cross-site parent document, see [whatwg/storage#142](https://github.com/whatwg/storage/issues/142#issuecomment-1122147159). In practice, this might involve comparing the [=site for cookies=] or performing a [=same site=] check with the top-level document.
 
-1. [=Queue a global task=] on the [=permissions task source=] given |global| to [=/resolve=] |p| with |global|'s [=environment/has storage access=].
+1. If |global|'s [=environment/has storage access=] is true, [=/resolve=] |p| with true and return |p|.
+1. Run the following steps [=in parallel=]:
+    1. Let |has unpartitioned data access| be the result of whether the user agent allows |doc| access [=unpartitioned data=] based on [=global user settings=].
```

Comment:
I think we should look at whether the user agent is configured to allow access to cookies for the (top-level site, embedded site) combination. Or something like that, perhaps it needs to be implementation-defined.